src/main/java/cinema/service/AuthenticationServiceImpl.java

```diff
@@ -27,6 +29,9 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        User userAdd = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(userAdd);
+        return userAdd;
```

Comment:
You can use the same reference `user` without even reassigning it

Reply:
was getting errors otherwise, but now seems ok